h/views/api/helpers/links.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+"""Helpers for generating and retrieving service link metadata"""
+
+from __future__ import unicode_literals
+
+
+def register_link(link, versions, registry):
+    """
+    Register an API service's metadata for its supported versions
+
+    Add an entry for the indicated ``link`` onto the ``registry`` for each
+    of the versions it claims to supported. These entries include basic
+    endpoint metadata like description and primary HTTP method.
+
+    These data structures are later referenced to deliver an index of our
+    API endpoints at `GET /api/`.
+
+    :arg link: Service metadata
+    :type link: dict
```

Comment:
As noted elsewhere, I think we should consider creating an actual class for links in future to better document the various fields they should have.